### PR TITLE
[Issue 2668] Setup local oauth2 server for API to connect to

### PR DESCRIPTION
**Pull request by babebe**
## Summary
Fixes #{[2668](https://github.com/HHS/simpler-grants-gov/issues/2668)}

## Changes proposed
Added Mock Oauth2 Server to api/docker-compose.yml 
Added custom config.json file with one test user 
Added custom mock-server-key.json file with one jwks key

## Context for reviewers
Pull mock server image specified in docker compose file and run it(Can also use `make start`) 
Can hit `http://localhost:5000/default/debugger` successfully 
`http://localhost:5000/issuer1/.well-known/openid-configuration` lists endpoints you can hit.

### CODE REVIEW

File: api/config.json
Comment (chouinar) [inline api/config.json:1]: Could we put this file and the mock-server-key.json into a folder within this directory?

Something like `mock-oauth` for the folder?

**Review by chouinar — COMMENTED** [review]
Just an organizational comment, if there are any questions I still have, we'll sort those out as we get to them in the usage of this

**Review by emilycnava — COMMENTED** [review]
Had to change port to 5001 on my machine as 5000 was colliding with something else, but aside from the port number issue both endpoints were reachable!

File: api/config.json
Comment (babebe) [reply api/config.json:1]: Yes! That would be better! Done

File: api/docker-compose.yml
[the remark this answers is not in the document]
Comment (babebe) [reply api/docker-compose.yml:1]: updated to 5001

**Review by chouinar — APPROVED** [review]
LGTM!

**Pull request merged**